libraries/botbuilder/src/teamsActivityHandler.ts

```diff
@@ -606,6 +606,88 @@ export class TeamsActivityHandler extends ActivityHandler {
         }
     }
 
+    /**
+     * Override this method to change the dispatching of MessageUpdate activities.
+     *
+     * @param context A context object for this turn.
+     * @returns A promise that represents the work queued.
+     */
+    protected async dispatchMessageUpdateActivity(context: TurnContext): Promise<void> {
+        if (context.activity.channelId == 'msteams') {
+            const channelData = context.activity.channelData as TeamsChannelData;
+
+            switch (channelData.eventType) {
+                case 'undeleteMessage':
+                    return await this.onTeamsMessageUndelete(context);
```

Comment:
@singhk97 is this functionality live already? Why is the `"undeleteMessage"` sub-type coming in under a `"messageUpdate"` activity? It feels more intuitive to arrive as a subtype under the `"messageDelete"` activity.

Reply:
This matches the Teams messaging and is consistent with the DotNet version.